pkg/morserino_com/listen_com_test.go

```diff
@@ -0,0 +1,130 @@
+package morserino_com
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"testing/iotest"
+
+	"go.bug.st/serial/enumerator"
+)
+
+func TestGet_MorserinoPort_automatically(t *testing.T) {
+	type args struct {
+		genericEnumPorts comPortEnumerator
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			"Happy case",
+			args{
+				mockPortEnumerator{ports: []*enumerator.PortDetails{
+					{Name: "COM Port Name", IsUSB: true, VID: "XXXX", PID: "YYYY", SerialNumber: "001"},
+					{Name: "Morserino Port", IsUSB: true, VID: "10C4", PID: "EA60", SerialNumber: "002"}},
+				},
+			},
+			"Morserino Port", false,
+		},
+		{
+			"No morserino detected",
+			args{
+				mockPortEnumerator{ports: []*enumerator.PortDetails{
+					{Name: "COM Port Name", IsUSB: true, VID: "XXXX", PID: "YYYY", SerialNumber: "001"}},
+				},
+			},
+			"", true,
+		},
+		{
+			"Too many Morserinos detected",
+			args{
+				mockPortEnumerator{ports: []*enumerator.PortDetails{
+					{Name: "COM Port Name", IsUSB: true, VID: "XXXX", PID: "YYYY", SerialNumber: "001"},
+					{Name: "Morserino Port", IsUSB: true, VID: "10C4", PID: "EA60", SerialNumber: "002"},
+					{Name: "Morserino Port2", IsUSB: true, VID: "10C4", PID: "EA60", SerialNumber: "003"}},
+				},
+			},
+			"", true,
+		},
+		{
+			"Port enumeration went wrong",
+			args{mockPortEnumerator{ports: nil, err: fmt.Errorf("An error occured")}},
+			"", true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := Get_MorserinoPort_automatically(tt.args.genericEnumPorts)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Get_MorserinoPort_automatically() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("Get_MorserinoPort_automatically() = %v, want %v", got, tt.want)
+			}
```

Comment:
If you declare the error you expect as an `error` in your test case, you would be able to write.
Also, please stick to using `stretchr/testify` :pray:, which makes the test code way easier to read.

```suggestion
			got, err := Get_MorserinoPort_automatically(tt.args.genericEnumPorts)
			assert.Equal(t, tt.wantErr, err)
			assert.Equal(t, tt.want, got)
```

Reply:
Very good point. This relates to your advise of steering clear of generated code (the case for these tests) until proficient enough to have a critical eye on the "artificial" code.

Reply:
Looking a little closer at this: It doesn't require a lot of extra typing. It makes easier/more readable to do test driven development. The comment above is very valid: as an apprentice, I need to train by painting the fence or catching a fly...